Slim/Utils/Scanner/Remote.pm

```diff
@@ -703,6 +699,118 @@ sub parseAACHeader {
 	$cb->( $track, undef, @{$pt} );
 }
 
+sub parseMp4Header {
+	my ( $http, $dataref, $track, $args, $url ) = @_;
+
+	Slim::Formats->loadTagFormatForType('mp4');
+	my $formatClass = Slim::Formats->classForFormat('mp4');
+	my $info = $formatClass->parseStream($dataref, $args);
+	
+	if ( ref $info ne 'HASH' ) {
+		if ( !$info ) {
+			$log->error("unable to parse mp4 header");
+			$args->{cb}->( $track, undef, @{$args->{pt} || []} );
+			return 0;
+		} elsif ( $info < 0 ) {
+			return 1;
+		} else {
+			my $query = Slim::Networking::Async::HTTP->new;
+			$http->disconnect;
+	
+			# re-calculate header all the time (i.e. can't go direct at all)
+			$track->initial_block_type(Slim::Schema::RemoteTrack::INITIAL_BLOCK_ALWAYS);
+		
+			main::INFOLOG && $log->is_info && $log->debug("'mdat' reached before 'moov' at ", length($args->{_scanbuf}), " => seeking with $args->{_range}");
+	
+			$query->send_request( {
+				request     => HTTP::Request->new( GET => $url,  [ 'Range' => "bytes=$info-" ] ),
+				onStream  	=> \&parseMp4Header,
+				onError     => sub {
+				my ($self, $error) = @_;
+					$log->warn( "could not find MP4 header $error" );
+					$args->{cb}->( $track, undef, @{$args->{pt} || []} );
+				},
+				passthrough => [ $track, $args, $url ],			
+			} );
+	
+			return 0;
+		}	
+	} 
+
+	my ($samplesize, $channels);
+	my $samplerate = $info->{samplerate};
+	my $duration = $info->{song_length_ms} / 1000;
+	my $bitrate = $info->{avg_bitrate};
+	
+	if ( my $item = $info->{tracks}->[0] ) {
+		my $format;
+		
+		$samplesize = $item->{bits_per_sample};
+		$channels = $item->{channels};
+
+		# If encoding is alac, the file is lossless
+		if ( $item->{encoding} && $item->{encoding} eq 'alac' ) {
+			$format = 'alc';	
+			# bitrate will be wrong b/c we only gave a header, not a real file
+			$bitrate = $duration ? $track->audio_size * 8 / $duration : 850_000;
+		} elsif ( $item->{encoding} && $item->{encoding} eq 'drms' ) {
+			$track->drm(1);
+		} 
+		
+		# Check for HD-AAC file, if the file has 2 tracks and AOTs of 2/37
+		if ( defined $item->{audio_object_type} && (my $item2 = $info->{tracks}->[1]) ) {
+			if ( $item->{audio_object_type} == 2 && $item2->{audio_object_type} == 37 ) {
+				$samplesize   = $item2->{bits_per_sample};
+				$format = 'sls';
+			}
+		}
+		
+		# use process_audio hook & format is set by parser
+	    # MPEG-4 audio = 64,  MPEG-4 ADTS main = 102, MPEG-4 ADTS Low Complexity = 103
+		# MPEG-4 ADTS Scalable Sampling Rate = 104
+		if ( $info->{audio_initiate} ) {
+			$track->initial_block_type(Slim::Schema::RemoteTrack::INITIAL_BLOCK_ALWAYS);
+			$track->audio_initiate($info->{audio_initiate});
+			$format = $info->{audio_format};
+		} elsif ( !$format && $item->{audio_type} ~~ 102..104 ) {
```

Comment:
I put it there because I've never seen a version of LMS (I needed to check that for my other plugins) > 7.9 that does not at least have 5.14

Reply:
Just saw that Perl is warning about the smartmatch being experimental:
>Smartmatch is experimental at /Users/mh/SynologyDrive/git/server/Slim/Utils/Scanner/Remote.pm line 779.

Some sites recommend to use `use experimental 'smartmatch'` - but that didn't work on my 5.18. Others simply claim:

> but preferably don't use smartmatch at all, it was retroactively marked experimental because it has too many edge cases for comfort.

In this case it's basically replacing

```perl
$item->{audio_type} >= 102 && $item->{audio_type} <= 104
```

isn't it? Please get rid of it. Users will complain/ask about it.

Reply:
done